### PR TITLE
feat(text-area): add pf-text-area

### PR DESCRIPTION
**Pull request by bennypowers**
## What I did

1. add features to InternalsController
2. move code from PfTextInput to InternalsController
3. add pf-text-area

## Testing Instructions

1. extensive usability testing is required
2. re-test pf-text-area which is also updated here

## Notes to Reviewers

1. depending on which PR is merged first, this also could impact pf-checkbox, radio, switch - anything with element internals
2. i'm not 100% clear on the polyfill story, so try in webkit and firefox as well

### CODE REVIEW

File: core/pfe-core/controllers/internals-controller.ts [hunk 1]
```diff
@@ -46,75 +99,107 @@ export class InternalsController implements ReactiveController, ARIAMixin {
...
-
-  hostConnected?(): void
 
-  setFormValue(...args: Parameters<ElementInternals['setFormValue']>) {
-    return this.#internals.setFormValue(...args);
+    // for (const [key, val] of Object.entries(this.options ?? {})) {
+    //   if (isARIAMixinProp(key)) {
+    //     this[key] = val;
+    //   }
+    // }
```
Comment (zeroedin) [inline core/pfe-core/controllers/internals-controller.ts:168]: remove

File: elements/pf-text-area/pf-text-area.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+import { InternalsController } from '@patternfly/pfe-core/controllers/internals-controller.js';
+
+import styles from './pf-text-area.css';
+import { Logger } from '@patternfly/pfe-core/controllers/logger.js';
+
+/**
+ * A **text area** component is used for entering a paragraph of text that is longer than one line.
+ *
+ */
```
Comment (zeroedin) [inline elements/pf-text-area/pf-text-area.ts:14]: document css props

File: elements/pf-text-area/test/pf-text-area.spec.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    it('should upgrade', async function() {
+      element = await createFixture<PfTextArea>(html`<pf-text-area></pf-text-area>`);
+      const klass = customElements.get('pf-text-area');
+      expect(element)
+        .to.be.an.instanceOf(klass)
+        .and
+        .to.be.an.instanceOf(PfTextArea);
+    });
+  });
+});
```
Comment (zeroedin) [inline elements/pf-text-area/test/pf-text-area.spec.ts:21]: add tests

File: elements/pf-text-area/pf-text-area.ts [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+    return this.shadowRoot?.getElementById('textarea') as HTMLTextAreaElement ?? null;
+  }
+
+  override willUpdate() {
+    this.#derivedLabel = this.accessibleLabel || this.#internals.computedLabelText;
+  }
+
+  override render() {
+    return html`
+      <textarea id="textarea"
```
Comment (zeroedin) [inline elements/pf-text-area/pf-text-area.ts:196]: should set `.vertical` or `.horizontal` class to limit resize

File: elements/pf-text-area/pf-text-area.css [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+  border-color: var(--pf-c-form-control--disabled--BorderColor);
+}
+
+textarea[readonly] {
+  background-color: var(--pf-c-form-control--readonly--BackgroundColor);
+}
+
+.vertical { resize: vertical; }
+.horizontal { resize: horizontal; }
+
```
Comment (zeroedin) [inline elements/pf-text-area/pf-text-area.css:192]:
```Suggestion
.both { resize: both; }
```
The resize property takes a value of `both` and there is a patternfly modifier `.pf-m-resize-both`

File: elements/pf-text-area/test/pf-text-area.spec.ts (hunk 3 above)
Comment (bennypowers) [reply elements/pf-text-area/test/pf-text-area.spec.ts:97]: how much of this can we get away with deferring until after release?

**Review by zeroedin — APPROVED** [review]
Linguistic genius: text mastery

**Comment by bennypowers** [comment]
merging prior to UX testing. @hellogreg, @nikkimk please open new issues for anything you catch. Hopefully we can sort those before 2.5.0, otherwise we'll issue patches.

**Pull request merged**